src/track/trackmetadatataglib.cpp

```diff
@@ -63,47 +53,52 @@ static_assert(sizeof(wchar_t) == sizeof(QChar), "wchar_t is not the same size th
 
 namespace mixxx {
 
-namespace {
+namespace taglib {
 
-const QString kFileTypeAIFF("aiff");
-const QString kFileTypeFLAC("flac");
-const QString kFileTypeMP3("mp3");
-const QString kFileTypeMP4("mp4");
-const QString kFileTypeOggVorbis("ogg");
-const QString kFileTypeOggOpus("opus");
-const QString kFileTypeWAV("wav");
-const QString kFileTypeWavPack("wv");
+QDebug operator<<(QDebug debug, FileType fileType) {
+    return debug << static_cast<std::underlying_type<FileType>::type>(fileType);
+}
 
-// Preferred picture types for cover art sorted by priority
-const std::array<TagLib::ID3v2::AttachedPictureFrame::Type, 4> kPreferredID3v2PictureTypes = {
-        TagLib::ID3v2::AttachedPictureFrame::FrontCover, // Front cover image of the album
-        TagLib::ID3v2::AttachedPictureFrame::Media, // Image from the album itself
-        TagLib::ID3v2::AttachedPictureFrame::Illustration, // Illustration related to the track
-        TagLib::ID3v2::AttachedPictureFrame::Other
-};
-const std::array<TagLib::FLAC::Picture::Type, 4> kPreferredVorbisCommentPictureTypes = {
-        TagLib::FLAC::Picture::FrontCover, // Front cover image of the album
-        TagLib::FLAC::Picture::Media, // Image from the album itself
-        TagLib::FLAC::Picture::Illustration, // Illustration related to the track
-        TagLib::FLAC::Picture::Other
-};
+bool hasID3v1Tag(TagLib::MPEG::File& file) {
+#if (TAGLIB_HAS_TAG_CHECK)
+    return file.hasID3v1Tag();
+#else
+    return nullptr != file.ID3v1Tag();
+#endif
+}
 
-inline bool hasID3v2Tag(TagLib::MPEG::File& file) {
+bool hasID3v2Tag(TagLib::MPEG::File& file) {
 #if (TAGLIB_HAS_TAG_CHECK)
     return file.hasID3v2Tag();
 #else
     return nullptr != file.ID3v2Tag();
 #endif
 }
 
-inline bool hasAPETag(TagLib::MPEG::File& file) {
+bool hasAPETag(TagLib::MPEG::File& file) {
 #if (TAGLIB_HAS_TAG_CHECK)
     return file.hasAPETag();
 #else
     return nullptr != file.APETag();
 #endif
 }
 
+namespace {
```

Comment:
I've only exposed what is actually needed for the test. Of course, we can 
also expose the remaining functions for consistency.

On 28.08.2016 12:49, Daniel Schürmann wrote:
>
> In src/track/trackmetadatataglib.cpp 
> <https://github.com/mixxxdj/mixxx/pull/997#discussion_r76529737>:
>
> >  #if (TAGLIB_HAS_TAG_CHECK)
> >      return file.hasAPETag();
> >  #else
> >      return nullptr != file.APETag();
> >  #endif
> >  }
> >
> > +namespace {
>
> now we have this:
>
> |namespace mixxx { namespace taglib { hasAPETag() namespace { hasID3v2Tag() 
> } // anonymous namepace } // namespace taglib } // namespace mixxx Why not 
> treat these functions similar? |
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/997/files/a6f69573de0ced1d2bf83499679a94e0ab2925b9#r76529737>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-ahFeZsKkGDXYTcqsAjcAwEkgb1p5ks5qkWeogaJpZM4JustG>.
>